scipy/sparse/linalg/isolve/utils.py

```diff
@@ -120,4 +111,17 @@ def postprocess(x):
         if A.shape != M.shape:
             raise ValueError('matrix and preconditioner have different shapes')
 
+    # set initial guess
+    if x0 is None:
+        x = zeros(N, dtype=xtype)
+    elif x0 is True:  # use nonzero initial guess ``M * b``
+        bCopy = b.copy()
+        x = M.matvec(bCopy)
+    else:
+        x = array(x0, dtype=xtype)
+        if not (x.shape == (N,1) or x.shape == (N,)):
+            raise ValueError('shapes of A {} and x0 {} are incompatible'
+                            .format(A.shape, x.shape))
```

Comment:
F string would be more appropriate here, isn't it ?

Reply:
Some original codes in SciPy use `.format`, I don't change the usage because it is not the purpose of the PR. Is there any special reason to use the f-string instead of `.format`?

Reply:
the original format used to be `.format` because f-string didn't exist at that time. I found f-string more clean and readable. Since you are updating the code, I would think this is a nice add at the same time

Reply:
either way is fine